cluster/addons/rbac/cluster-autoscaler/cluster-autoscaler-rbac.yaml

```diff
@@ -0,0 +1,64 @@
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cluster-autoscaler
+  labels:
+    addonmanager.kubernetes.io/mode: Reconcile
+rules:
+  # leader election
+  - apiGroups: [""]
+    resources: ["endpoints"]
+    verbs: ["create"]
+  - apiGroups: [""]
+    resources: ["endpoints"]
+    resourceNames: ["cluster-autoscaler"]
```

Comment:
I assume the autoscaler is using this for leader locking? If so, as a future improvement, I'd recommend switching away from endpoints (possibly to a configmap, like the kube components have) to avoid continuously pushing unnecessary endpoint events to watchers (all the kube-proxy processes on all the nodes)

Reply:
Yes, it's for leader election and locking.
I thought that the algorithm (and object used) is the same as in kube-scheduler but I will have another look and create an enhancement PR in CA - thanks for the suggestion.

Reply:
Once we have Lease object: https://github.com/kubernetes/kubernetes/pull/64246/ we should migrate everyone to that. ConfigMap is better, but still not ideal, and since we are going to have API exactly for this purpose let's avoid updating it more than once.

Reply:
I've created [PR 944](https://github.com/kubernetes/autoscaler/issues/944) in the autoscaler repository.